src/storage/compactor/src/server.rs

```diff
@@ -93,7 +92,7 @@ pub async fn compactor_serve(
     let sstable_store = Arc::new(SstableStore::for_compactor(
         object_store,
         storage_config.data_directory.to_string(),
-        storage_config.block_cache_capacity_mb * (1 << 20),
+        1 << 20, // set 1MB memory to avoid panic.
         storage_config.meta_cache_capacity_mb * (1 << 20),
```

Comment:
`data_cache_capacity` and `meta_cache_capacity` don't make sense for sstable store used in compactor. I think we can have:
- `SstableStoreBase`: the implementation interacts with ObjectStore only.
- `SstableStore`: `SstableStoreBase` + block/meta/tiered cache.
- `CompactorSstableStore`: `SstableStoreBase` + memory limiter.

We can address this in a separate PR if this PR is urgent.

Reply:
ok. I will pull another PR to resolve this problem.

Reply:
IMO, we may introduce a trait named `SstableStoreRead`, providing methods such as `get` and `scan`. And SstableIterator can have a generic parameter of such trait bound, and all sstable related read can be performed via this trait

Reply:
Agree